src/vmm/src/lib.rs

```diff
@@ -1093,4 +1147,48 @@ mod tests {
             assert!(fdt.write_to_mem(&vmm.guest_memory, fdt_offset).is_err());
         }
     }
+    #[test]
+    fn test_address_alloc() {
+        let memory_config = MemoryConfig {
+            size_mib: MEM_SIZE_MIB,
+        };
+        #[cfg(target_arch = "x86_64")]
+        let start_addr = MMIO_GAP_START;
+        #[cfg(target_arch = "aarch64")]
+        let start_addr = AARCH64_MMIO_BASE;
+        let mut address_alloc = Vmm::create_address_allocator(&memory_config).unwrap();
+        // Trying to allocate at an address before the base address.
```

Comment:
nit: Can you add a white line between each test case? The code looks a bit crowded.